lib/ecto/changeset.ex

```diff
@@ -1072,7 +1072,7 @@ defmodule Ecto.Changeset do
       apply_changes(changeset)
 
   """
-  @spec apply_changes(t) :: Ecto.Schema.t
+  @spec apply_changes(t) :: Ecto.Schema.t | data
```

Comment:
@josevalim sorry, I'm not well-versed in typespecs, I was essentially cargo-culting it :)

I see data is defined as: 

```elixir
# https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/changeset.ex#L239
@type data :: map()
```

so `map` vs `map()` is different wrt `nil`?